src/domain/Ladder.java

```diff
@@ -0,0 +1,32 @@
+package domain;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class Ladder extends ShapeInitializer{
+    private List<LadderRow> ladderRows;
+    private List<NameTag> nametags;
+    private int numberOfPlayer;
+
+    public Ladder(String[] names, int ladderDepth) {
+        this.ladderRows = new ArrayList<>();
+        this.nametags = new ArrayList<>();
+        this.numberOfPlayer = names.length;
+
+        Arrays.stream(names)
+                .forEach(name -> nametags.add(new NameTag(name)));
+
+        IntStream.range(0, ladderDepth)
+                .forEach(row -> ladderRows.add(new LadderRow(numberOfPlayer)));
```

Comment:
```suggestion
        this.ladderRows = IntStream.range(0, ladderDepth)
                .map(x -> new LadderRow(numberOfPlayer))
                .collect(Collectors.toList());
        this.nametags = Arrays.stream(names)
                .map(NameTag::new)
                .collect(Collectors.toList());
        this.numberOfPlayer = names.length;
```

저라면 이렇게 작업해줬을 것 같고, 더 나아가서 생성 단계에서 이런 생성 로직들이 들어가는건 바람직하지 않다고도 생각합니다.
생성자는 정말 객체 생성만 하게끔 가볍게 관리하는게 어떨까 싶네요.